Develop Bubble Sort Algorithm

MarilenaRoque opened the pull request:
On this branch, we have create two methods

- bubble_sort: it sorts an array of numbers using the bubble sort technique
-bubble_sort_by: It sorts an array of strings using its lengths 

We have applied the theory learned about yield and blocks for both methods

MarilenaRoque committed:
bubble_sort method created

Yahir-Cardona committed:
bubble-sort_yield created

MarilenaRoque committed:
Added README.md and finish bubble_sort

Joyned bubble_sort and bubble_sort_by on a single version
Added README.md model

Yahir-Cardona committed:
readme created

Yahir-Cardona committed:
Linters added and fixed

MarilenaRoque committed:
Fixing README information

MarilenaRoque committed:
Added sticklers file

MarilenaRoque committed:
Fixing stickler file

shshamim63 commented on bubble_sort.rb:
```diff
@@ -0,0 +1,45 @@
+  end
+end
+
+def bubble_sort_by(new_array)
+  flag = 0
+  while flag.zero?
+    flag = 1
+    new_array.each_with_index do |numb, position|
+      next if position == new_array.length - 1
+
+      test = yield(numb, new_array[position + 1])
+      next unless test.positive?
+
+      new_array[position] = new_array[position + 1]
+      new_array[position + 1] = numb
+      flag = 0
+    end
+  end
+end
```
- Sort returns the sorted array. So your bubble_sort_by should do the same task. But it is not returning anything. That's why while i tried to execute the following code it did not work.
```ruby
print bubble_sort_by([1, 5, 3, 2, 1]) {|num1, num2| num1 - num2}
```
So, after line 32 you should place new_array.

shshamim63 commented on bubble_sort.rb:
```diff
@@ -0,0 +1,45 @@
+def bubble_sort(new_array)
+  flag = 0
+  while flag.zero?
+    flag = 1
+    new_array.each_with_index do |numb, position|
+      next if position == new_array.length - 1
+
+      test = yield(numb, new_array[position + 1])
+      next unless test.positive?
+
+      new_array[position] = new_array[position + 1]
+      new_array[position + 1] = numb
+      flag = 0
+    end
+  end
+end
```
- bubble_sort does not interact with any block. So yield is redundant. Instead, it should compare two items.
 - Sort returns the sorted array. So your bubble_sort should do the same task. But it is not returning anything. That's why while i tried to execute the following code it did not work.
```ruby
print bubble_sort([1, 5, 3, 2, 1])
```
So, after line 15 you should place new_array.

yahir91 replied:
We implemented the next changes

Added new_array to both methods, now it prints both results.
We fixed the redundant yield on bubble_sort method.

Thank you for your review.

shshamim63 reviewed (changes requested):
## Status :1234: Changes Required
___
Hi @MarilenaRoque and @yahir91 :wave: 
Almost completed the tasks. just need to cover some edge cases. Let's get to them.
___
### :spades: Mandatory Chagnes :spades: 
___
- Check the comments below ans resolve the issue
___
##### After resolving all the issues please send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

MarilenaRoque committed:
Fixing Return and yield

Added a return value for the bubble_sort method
Added a return value for the bubble_sort_by method
Remove the yield for the bubble_sort method

MarilenaRoque committed:
Fixed return issue

MarilenaRoque committed:
Format fixed

MarilenaRoque committed:
Extra blank line removed

Yahir-Cardona committed:
space added for a linter error

MarilenaRoque commented:
We have fixed all the issues pointed:

- [x] Both methods return the array
- [x] Removed yield for bubble_sort method

Thank you so much for the review @shshamim63!

serdg0 reviewed (approved):
**Congratulations** @MarilenaRoque @yahir91  :fireworks: :100: :fireworks:
Feel free to merge.
Keep the good work in your next project!
**Happy Coding!** :computer: :nerd_face:
**_Sergio Diaz_**

MarilenaRoque commented:
Thank you so much for the review @serdg0! :rocket:
